code/_core/obj/structure/totem.dm

```diff
@@ -323,3 +323,47 @@
 	icon_state = "shock"
 	chosenProjectile = /obj/projectile/magic/lightning_bolt
 	chosenDamageType = /damagetype/ranged/magic/lightning
+
+/obj/structure/totem/blood_heal
+	name = "totem of blood regeneration"
+	desc = "I'm melting, I'm melting!"
+	desc_extended = "A totem that will restore the caster's and allies blood."
+	icon_state = "blood"
+
+/obj/structure/totem/blood_heal/totemic_effect() //will need testing and help to balance this
+	var/turf/T = get_turf(src)
+	for(var/mob/living/L in viewers(4,T))
+		if(L.dead)
+			continue
+		if(L.loyalty_tag != affecting_faction)
+			continue
+		if(L.immortal)
+			continue
+		if(!istype(L.health))
+			continue
+		if(L.blood_volume + (5*leveled_effect) >= L.blood_volume_max)
+			continue
+		L.blood_volume += (5*leveled_effect)
```

Comment:
I think you should instead do a check if the blood volume is greater or equal to the blood volume max, and then make it so that the amount of blood added is ```min(blood_volume_max - blood_volume_current,leveled_effect*5)```